### ISSUE TITLE
M1 shader-tools

### ISSUE THREAD

**Issue by DanAlbert** [android/ndk#1577]
Subtask of https://github.com/android/ndk/issues/1299.

@dneto0 FYI, but I don't think any work is actually needed from your side. Pretty sure it's just a matter of getting the build servers in order and adding some flags.

**Comment by rprichard** [comment]
This NDK checkbuild.py patch appears sufficient to turn the shader-tools binaries into universal arm64+x86_64 binaries:
```
diff --git a/ndk/cmake.py b/ndk/cmake.py
index 7290de9f..ddcbc26d 100644
--- a/ndk/cmake.py
+++ b/ndk/cmake.py
@@ -35,7 +35,7 @@ SYSTEM_NAME_MAP = {
 }
 
 HOST_TRIPLE_MAP = {
-    Host.Darwin: 'x86_64-apple-darwin',
+    Host.Darwin: 'apple-darwin',
     Host.Linux: 'x86_64-linux-gnu',
     Host.Windows64: 'x86_64-w64-mingw32',
 }
@@ -179,6 +179,8 @@ class CMakeBuilder:
         }
         if self.host.is_windows:
             defines['CMAKE_RC'] = str(self.toolchain.rescomp)
+        if self.host == Host.Darwin:
+            defines['CMAKE_OSX_ARCHITECTURES'] = 'x86_64;arm64'
         return defines
 
     def clean(self) -> None:
```
I wasn't sure what to do with the triple, so I tried just cutting x86_64 off of it. Maybe we should omit the `CMAKE_C_COMPILER_TARGET` and `CMAKE_CXX_COMPILER_TARGET` defines instead.

I'm running on macOS 11.5.2 with Xcode 12.5.1, though. I think the aosp-master-ndk buildbot is too old. The aosp-llvm-toolchain buildbot is new enough, though?

**Comment by rprichard** [comment]
https://android-review.googlesource.com/c/platform/ndk/+/1817588

**Comment by DanAlbert** [comment]
Should be fixed in r23 [build 8486889](https://ci.android.com/builds/branches/aosp-ndk-release-r23/grid?head=8486889&tail=8486889).
